src/components/StackedChart/index.jsx

```diff
@@ -52,12 +59,8 @@ const StackedChart = ({ title, type, data, isCurrency, showTimeFilter, maxHeight
           currentStackedDataValue += series.value;
         });
 
-      // get the previous day data by subtracting
-      // one day from the active label
-      // (the label is the actual date in the YYYY-MM-DD format)
-      const oneDayOldDate = new Date(activeLabel);
-      oneDayOldDate.setDate(oneDayOldDate.getDate() - 1);
-      const oneDayOldData = filteredData.find((data) => data.time === dayjs(oneDayOldDate).format('YYYY-MM-DD'));
+      // get the previous day data
+      const oneDayOldData = filteredData[activeIndex !== 0 ? activeIndex - 1 : activeIndex];
```

Comment:
Negative logic is harder to parse
```suggestion
      const oneDayOldData = filteredData[activeIndex === 0 ? activeIndex : activeIndex - 1];
```